lib/fluent/plugin/out_secondary_file.rb

```diff
@@ -61,34 +61,40 @@ def configure(conf)
 
       @dir_perm = system_config.dir_permission || Fluent::DEFAULT_DIR_PERMISSION
       @file_perm = system_config.file_permission || Fluent::DEFAULT_FILE_PERMISSION
+      @need_worker_lock = system_config.workers > 1
+      @need_thread_lock = @primary_instance.buffer_config.flush_thread_count > 1
     end
 
     def multi_workers_ready?
-      ### TODO: add hack to synchronize for multi workers
       true
     end
 
+    def start
+      super
+      extend WriteLocker
```

Comment:
I'm not sure how I should handle these Ruby features...
What I thought was as follows.

* I just wanted to clean up the methods by combining some specific function methods into a module.
* The `WriteLocker` module depends on the instance members of `SecondaryFileOutput`.
* So I felt uncomfortable with the form in which `SecondaryFileOutput` inherits `WriteLocker`.
  * I thought the dependency was reversed.

Reply:
Hmm, should I just write the methods directly under the `SecondaryFileOutput`, not using `module`?

Reply:
In this case I prefer not using module since there aren't so many code in this plugin for now.

BTW methods in `WriteLocker` might be useful for other output plugins, so place them just under `Fluent::Plugin::Output` might be better.

Reply:
Indeed! I will try it!

Reply:
Fixed at ce32ac57a07121d69adf30d3b543ea11041ce613.